metron-platform/metron-parsing/metron-parsers/src/main/java/org/apache/metron/parsers/leef/LEEFParser.java

```diff
@@ -0,0 +1,236 @@
+	private Pattern pext;
+
+	public void init() {
+
+		// CEF Headers: Device Vendor|Device Product|Device Version|Device Event
+		// Class ID|Name|Severity
+
+		String syslogTime = "(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:tember)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)\\b +(?:(?:0[1-9])|(?:[12][0-9])|(?:3[01])|[1-9]) (?!<[0-9])(?:2[0123]|[01]?[0-9]):(?:[0-5][0-9])(?::(?:(?:[0-5]?[0-9]|60)(?:[:.,][0-9]+)?))(?![0-9])?";
+		String syslogTime5424 = "(?:\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(?:\\.\\d+)?(?:Z|[+-]\\d{2}:\\d{2}))";
+		String syslogPriority = "<(?:[0-9]+)>";
+		String syslogHost = "[a-z0-9\\.\\\\-_]+";
+
+		StringBuilder sb = new StringBuilder("");
+		sb.append("(?<syslogPriority>");
+		sb.append(syslogPriority);
+		sb.append(")?");
+		sb.append("(?<syslogTime>");
+		sb.append(syslogTime);
+		sb.append("|");
+		sb.append(syslogTime5424);
+		sb.append(")?");
+
+		sb.append("(?<syslogHost>");
+		sb.append(syslogHost);
+		sb.append(")?");
+
+		sb.append(".*");
+
+		sb.append("LEEF:(?<Version>1.0|2.0|0)?\\|");
+
+		headerBlock("DeviceVendor", sb);
+		sb.append("\\|");
+		headerBlock("DeviceProduct", sb);
+		sb.append("\\|");
+		headerBlock("DeviceVersion", sb);
+		sb.append("\\|");
+		headerBlock("DeviceEvent", sb);
+		sb.append("\\|");
+		
+		// add optional delimiter header (only applicable for LEEF 2.0)
+		sb.append("(");
+		headerBlock("Delimiter", sb);
+		sb.append("\\|");
+		sb.append(")?");
+		
+		// extension capture:
+		sb.append(" ?(?<extensions>.*)");
+		String pattern = sb.toString();
+
+		p = Pattern.compile(pattern);
+
+		pext = Pattern.compile(EXTENSION_CAPTURE_PATTERN);
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<JSONObject> parse(byte[] rawMessage) {
+		List<JSONObject> messages = new ArrayList<>();
+
+		String cefString = new String(rawMessage, UTF_8);
+
+		Matcher matcher = p.matcher(cefString);
+
+		while (matcher.find()) {
+			JSONObject obj = new JSONObject();
+			if (!matcher.matches()) {
+				break;
+			}
+			LOG.debug("Found %d groups", matcher.groupCount());
+			obj.put("DeviceVendor", matcher.group("DeviceVendor"));
+			obj.put("DeviceProduct", matcher.group("DeviceProduct"));
+			obj.put("DeviceVersion", matcher.group("DeviceVersion"));
+			obj.put("DeviceEvent", matcher.group("DeviceEvent"));
+			
+			String ext = matcher.group("extensions");
+
+			// In LEEF 2.0 the delimiter can be specified
+			String version = matcher.group("Version");
+			if (version.equals("2.0")) {
+				String delimiter = matcher.group("Delimiter");
+				if (delimiter == null || delimiter.length() == 0) {
+					delimiter = "\\t";
+				}
+				delimiter = "(?<!\\\\)[" + delimiter.replace("^", "\\^").replace("\t", "\\t") + "]";
+				
+				String[] kvs = ext.split(delimiter); 
+				for (String kv: kvs) {
+					String[] a = kv.split("=");
+					obj.put(a[0], a[1]);
+				}
+			} else if (version.equals("1.0") || version.isEmpty()) {
+				// TODO - Support LEEF 1.0 delimiters
```

Comment:
Is this still "TODO" or is the tab char the delim?

Reply:
Nope. I did it and forgot to remove the comment. Done.